File: client/lib/models/post.dart
import 'package:json_annotation/json_annotation.dart';
part 'post.g.dart';

@JsonSerializable()
class Post {
  Post(this.name, this.description, this.image, this.tags, this.location,
      this.expiration, this.creator, this.id);

  String name;
  String description;
  String image;
  String location;
  String creator;
  @JsonKey(name: '_id')
  String id;
  DateTime expiration;
  List<String> tags;

  factory Post.fromJson(Map<String, dynamic> json) => _$PostFromJson(json);
  Map<String, dynamic> toJson() => _$PostToJson(this);
}

File: server/src/models/post.js
'use strict';

var mongoose = require('mongoose');
var ObjectId = mongoose.ObjectId;
var Schema = mongoose.Schema;

var PostSchema = new Schema({
  name: {
    type: String,
    required: true
  },
  creator: {
    type: ObjectId,
    required: true
  },
  description: {
    type: String,
    required: true
  },
  image: {
    type: String,
    required: true
  },
  location: {
    type: String,
    required: true
  },
  tags: {
    type: [String],
    required: true
  },
  expiration: {
    type: Date,
    required: true
  }
});

module.exports = mongoose.model('Post', PostSchema);
